connectors/sources/outlook.py

```diff
@@ -0,0 +1,944 @@
+    "recurrence",
+    "last_modified_time",
+    "subject",
+    "start",
+    "end",
+    "location",
+    "organizer",
+    "body",
+    "has_attachments",
+    "attachments",
+]
+
+END_SIGNAL = "FINISHED"
+
+
+def ews_format_to_datetime(source_datetime, timezone):
+    """Change datetime format to user account timezone
+    Args:
+        datetime: Datetime in UTC format
+        timezone: User account timezone
+    Returns:
+        Datetime: Date format as user account timezone
+    """
+    if isinstance(source_datetime, exchangelib.ewsdatetime.EWSDateTime):
+        return (source_datetime.astimezone(pytz.timezone(str(timezone)))).strftime(
+            "%Y-%m-%dT%H:%M:%SZ"
+        )
+    elif isinstance(source_datetime, exchangelib.ewsdatetime.EWSDate) or isinstance(
+        source_datetime, date
+    ):
+        return source_datetime.strftime("%Y-%m-%d")
+    else:
+        return source_datetime
+
+
+class TokenFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch authentication token from Graph API"""
+
+    pass
+
+
+class UsersFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch users from Active Directory"""
+
+    pass
+
+
+class UnauthorizedException(Exception):
+    """Exception class unauthorized calls"""
+
+    pass
+
+
+class Forbidden(Exception):
+    pass
+
+
+class NotFound(Exception):
+    pass
+
+
+class SSLFailed(Exception):
+    pass
+
+
+class RootCAAdapter(requests.adapters.HTTPAdapter):
+    """Class to verify SSL Certificate for Exchange Servers"""
+
+    def __init__(self):
+        self.ssl_verify = False
+        self.ssl_ctx = None
+
+    def cert_verify(self, conn, url, ssl_certificate_file, cert):
+        try:
+            super().cert_verify(
+                conn=conn, url=url, verify=self.ssl_verify, cert=self.ssl_ctx
+            )
+        except Exception as exception:
+            raise SSLFailed(
+                f"Something went wrong while verifying SSL certificate. Error: {exception}"
+            ) from exception
+
+
+class ExchangeUsers:
+    """Fetch users from Exchange Active Directory"""
+
+    def __init__(self, ad_server, domain, exchange_server, user, password, ssl_enabled):
+        self.ad_server = Server(host=ad_server)
+        self.domain = domain
+        self.exchange_server = exchange_server
+        self.user = user
+        self.password = password
+        self.ssl_enabled = ssl_enabled
+
+    @cached_property
+    def _create_connection(self):
+        return Connection(
+            server=self.ad_server,
+            user=self.user,
+            password=self.password,
+            client_strategy=SAFE_SYNC,
+            auto_bind=True,  # pyright: ignore
+        )
+
+    async def get_users(self):
+        warnings.filterwarnings("default")
+        ldap_domain_name_list = ["DC=" + domain for domain in self.domain.split(".")]
+        search_query = ",".join(map(str, ldap_domain_name_list))
```

Comment:
Why do you need `map` here?

Reply:
Oh Yes! I think map is not required. Thanks!